app/src/utils/datetime.ts

```diff
@@ -11,3 +17,46 @@ export const combineDateTime = (date: string, time?: string | null) => {
   }
   return new Date(`${date}T00:00:00`).toISOString();
 };
+
+/**
+ * Formats the time difference between two timestamps into a human-readable string.
+ *
+ * @param {string} startDate
+ * @param {string | null} startTime
+ * @param {string} endDate
+ * @param {string | null} endTime
+ * @returns {string | null} A formatted string indicating an amount of time
+ */
+export const formatTimeDifference = (
+  startDate: string,
+  startTime: string | null,
+  endDate: string,
+  endTime: string | null
+) => {
+  const startDateTime = startTime ? dayjs(`${startDate} ${startTime}`) : dayjs(startDate);
+  const endDateTime = endTime ? dayjs(`${endDate} ${endTime}`) : dayjs(endDate);
+
+  // Calculate the difference in milliseconds
+  const diff = endDateTime.diff(startDateTime);
+
+  // Calculate years, days, hours, minutes, and seconds
+  const years = Math.floor(diff / (1000 * 60 * 60 * 24 * 365));
```

Comment:
suggestion: You can use an additional param on diff to get specific date differences.
[DayJS Diff](https://day.js.org/docs/en/display/difference)

```js
const days = endDateTime.diff(startDateTime, 'd'); // 123
```

Reply:
oh gosh, that simplifies this. Thanks!